src/facade/dragonfly_connection.cc

```diff
@@ -754,6 +757,8 @@ void Connection::HandleRequests() {
       std::ignore =
           socket_->Write(io::Buffer("-ERR Bad TLS header, double check "
                                     "if you enabled TLS for your client.\r\n"));
+      stats_->tls_accept_disconnects++;
```

Comment:
From my understanding the handshake might have not started. Either way this is good for now - I will see if it can be improved.